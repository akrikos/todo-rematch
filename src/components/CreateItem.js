import * as React from "react";
import * as ReactRedux from "react-redux";
import * as Models from "../state_management/models.js"

class CreateItem extends React.Component {
  constructor() {
    super();
    this.state = { itemText: "" };
    this.createItem = this.createItem.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.setState({ itemText: event.target.value });
  }

  render() {
    return (
      <div>
        <form onSubmit={this.createItem}>
          <input
            className="hxTextCtrl"
            type="text"
            name="todo-item"
            value={this.state.itemText}
            onChange={this.handleChange}
          />
          <button className="hxBtn hxPrimary">Create</button>
        </form>
      </div>
    );
  }

  createItem(event) {
    this.props.addItem(new Models.TodoItem(this.state.itemText));
    this.setState({ itemText: '' });
    event.preventDefault();
  }
}

const createMapStateToProps = () => {
  return {};
};

const createMapDispatchToProps = dispatch => ({
  addItem: dispatch.todoItems.addItem
});

const ConnectedCreateItem = ReactRedux.connect(
  createMapStateToProps,
  createMapDispatchToProps
)(CreateItem);

export {
  CreateItem,
  ConnectedCreateItem
};