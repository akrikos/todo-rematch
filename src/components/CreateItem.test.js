import React from "react";
import ReactDOM from "react-dom";
import * as CreateItem from "./CreateItem";
import { shallow } from "enzyme";

let testItem;
const testAddItem = (item) => {
  testItem = item;
};

it("renders create item form structure as expected", () => {
  expect(<CreateItem.CreateItem addItem={function() {}} />).toMatchSnapshot();
});

it("adds an item on submission of the form", () => {
  
});
