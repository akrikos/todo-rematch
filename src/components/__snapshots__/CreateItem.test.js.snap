// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders create item form structure as expected 1`] = `
<CreateItem
  addItem={[Function]}
/>
`;
